### PR TITLE
Supports CDN on response

### LINKED ISSUE #466: Make amp-form samples work via AMP Cache
https://cdn.ampproject.org/c/s/ampbyexample.com/components/amp-form/ returns errors.

//cc @ithinkihaveacat

### PR DESCRIPTION
**Pull request by kul3r4**
This fixes #466 
I followed https://github.com/ampproject/amphtml/blob/master/spec/amp-cors-requests.md as reference.
@sebastianbenz PTAL

### CODE REVIEW

File: backend/request.go [hunk 1]
```diff
@@ -51,8 +51,10 @@ func isFormPostRequest(method string, w http.ResponseWriter) bool {
 }
 
 func EnableCors(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Origin", "https://cdn.ampproject.org")
```
Comment (sebastianbenz) [inline backend/request.go]: Can you please run the go formatter.

**Comment by kul3r4** [comment]
@sebastianbenz PTAL
I tried here: https://cdn.ampproject.org/c/s/chiara-ampbyexample.appspot.com/components/amp-form/ (responsive version only, as I changed manually the host to chiara-ampbyexample.com just for this page)

**Comment by ithinkihaveacat** [comment]
So every CDN needs to be whitelisted? Is there an official list of CDNs that anyone using amp-form can use? Is there any way to dynamically update?

/cc @pietergrayling

**Pull request merged**